tests/unit/apps/test_auxiliary.py

```diff
@@ -371,27 +373,29 @@ def test_auxiliary_app_unknown_version_raise_ApplicationError(model):
     exp_msg = f"'{charm}' with workload version {version} has no compatible OpenStack release."
 
     machines = {"0": MagicMock(spec_set=Machine)}
+    app = RabbitMQServer(
+        name=charm,
+        can_upgrade_to="3.8/stable",
+        charm=charm,
+        channel="3.8/stable",
+        config={"source": {"value": "distro"}},
+        machines=machines,
+        model=model,
+        origin="ch",
+        series="focal",
+        subordinate_to=[],
+        units={
+            f"{charm}/0": Unit(
+                name=f"{charm}/0",
+                workload_version=version,
+                machine=machines["0"],
+            )
+        },
+        workload_version=version,
+    )
+
     with pytest.raises(ApplicationError, match=exp_msg):
-        RabbitMQServer(
-            name=charm,
-            can_upgrade_to="3.8/stable",
-            charm=charm,
-            channel="3.8/stable",
-            config={"source": {"value": "distro"}},
-            machines=machines,
-            model=model,
-            origin="ch",
-            series="focal",
-            subordinate_to=[],
-            units={
-                f"{charm}/0": Unit(
-                    name=f"{charm}/0",
-                    workload_version=version,
-                    machine=machines["0"],
-                )
-            },
-            workload_version=version,
-        )
+        app._check_channel()
```

Comment:
We should be more specific in the function that raises this error and also change the test description to something like:

"""Test auxiliary application with unknown workload version."""

```suggestion
        app.get_latest_os_version()
```

Reply:
I do not understand why you want to change it to `get_latest_os_version`? That function is checking and unit, so it required unit as input argument.

Reply:
It's because that exp_message comes from this function. Using `_check_channel` eventually will call it, but it's a little bit misleading.

```
______________________________________________________________ test_auxiliary_app_unknown_version_raise_ApplicationError _______________________________________________________________

model = <AsyncMock spec_set='Model' id='138933752121424'>

    def test_auxiliary_app_unknown_version_raise_ApplicationError(model):
        """Test auxiliary upgrade plan with unknown version."""
        version = "80.5"
        charm = "rabbitmq-server"
        exp_msg = f"'{charm}' with workload version {version} has no compatible OpenStack release."
    
        machines = {"0": MagicMock(spec_set=Machine)}
        app = RabbitMQServer(
            name=charm,
            can_upgrade_to="3.8/stable",
            charm=charm,
            channel="3.8/stable",
            config={"source": {"value": "distro"}},
            machines=machines,
            model=model,
            origin="ch",
            series="focal",
            subordinate_to=[],
            units={
                f"{charm}/0": Unit(
                    name=f"{charm}/0",
                    workload_version=version,
                    machine=machines["0"],
                )
            },
            workload_version=version,
        )
>       app._check_channel()

tests/unit/apps/test_auxiliary.py:396: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
cou/apps/base.py:747: in _check_channel
    or self.is_valid_track(self.channel)
cou/apps/auxiliary.py:58: in is_valid_track
    (self.charm, self.series, self.current_os_release.codename), []
cou/apps/base.py:189: in current_os_release
    return min(self.os_release_units.keys())
cou/apps/base.py:240: in os_release_units
    os_version = self.get_latest_os_version(unit)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = RabbitMQServer(name='rabbitmq-server', can_upgrade_to='3.8/stable', charm='rabbitmq-server', channel='3.8/stable', con...4'>, workload_version='80.5')}, workload_version='80.5', packages_to_hold=None, wait_timeout=1800, wait_for_model=True)
unit = Unit(name='rabbitmq-server/0', machine=<MagicMock spec_set='Machine' id='138933746988464'>, workload_version='80.5')

    def get_latest_os_version(self, unit: Unit) -> OpenStackRelease:
        """Get the latest compatible OpenStack release based on the unit workload version.
    
        :param unit: Unit
        :type unit: Unit
        :return: The latest compatible OpenStack release.
        :rtype: OpenStackRelease
        :raises ApplicationError: When there are no compatible OpenStack release for the
                                  workload version.
        """
        compatible_os_versions = OpenStackCodenameLookup.find_compatible_versions(
            self.charm, unit.workload_version
        )
        if not compatible_os_versions:
>           raise ApplicationError(
                f"'{self.name}' with workload version {unit.workload_version} has no "
                "compatible OpenStack release."
            )
E           cou.exceptions.ApplicationError: 'rabbitmq-server' with workload version 80.5 has no compatible OpenStack release.

cou/apps/base.py:273: ApplicationError
```

Reply:
I change it to use `get_latest_os_version`. Thanks